### PR TITLE
Update `klaw-sync` module interface from v1.0 to v2.0.0 + 

### PR DESCRIPTION
**Pull request by luozhihua**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/manidlou/node-klaw-sync/blob/v2.0.0/CHANGELOG.md>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/klaw-sync/index.d.ts [hunk 1]
```diff
@@ -1,40 +1,62 @@
-// Type definitions for klaw-sync 1.1
+// Type definitions for klaw-sync 3.0
```
Comment (shiftkey) [inline types/klaw-sync/index.d.ts]: Does this need to go up to `3.0`? Can it stay on `2.0` to be in sync with the version of the dependency project?

File: types/klaw-sync/index.d.ts [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
+
+    type Filter = (item: Item) => boolean
+
+    interface Options {
+        /**
+         * @description any paths or `micromatch` patterns to ignore.
+         *              For more information on micromatch patterns: https://github.com/jonschlinkert/micromatch#features
+         * @deprecated v2.0.0
+         */
+        ignore?: string | string[]
```
Comment (shiftkey) [inline types/klaw-sync/index.d.ts]: If this has been deprecated in `2.0` then I'd just remove the parameter so that it's clear what's currently supported.

**Review by shiftkey — CHANGES_REQUESTED** [review]

File: types/klaw-sync/klaw-sync-tests.ts [hunk 3]
```diff
@@ -12,9 +12,20 @@ const defaultOptions = {}
...
-  nodir: true,
-  nofile: false,
+    ignore: ['.exe'],
+    nodir: true,
+    nofile: false,
 }
 
 klawSync('/some/dir', options).forEach(outputMessage)
+
+const optionsV2 = {
```
Comment (shiftkey) [inline types/klaw-sync/klaw-sync-tests.ts]: I'd just refresh the old `options` hash above to reflect what's currently supported.

File: types/klaw-sync/index.d.ts (hunk 2 above)
Comment (shiftkey) [reply types/klaw-sync/index.d.ts]: Let's focus on the v2 changes here, it seems like v3 is a smoother upgrade that we can follow up with later.

File: types/klaw-sync/klaw-sync-tests.ts [hunk 4]
```diff
@@ -12,9 +12,12 @@ const defaultOptions = {}
 klawSync('/some/dir', defaultOptions).forEach(outputMessage)
 
 const options = {
-  ignore: ['.exe'],
-  nodir: true,
-  nofile: false,
+    nodir: true,
+    nofile: false,
+    noRecurseOnFailedFilter: false,
+    filter(item: any) {
```
Comment (shiftkey) [inline types/klaw-sync/klaw-sync-tests.ts]: You can specify `item: klawSync.item` here instead of `item: any`

**Review by shiftkey — CHANGES_REQUESTED** [review]
One more bit of tidy up!

**Comment by luozhihua** [comment]
@shiftkey Thanks for help me, this is my first PR of DefinitelyTyped, your reviews is so helped.

**Comment by shiftkey** [comment]
@luozhihua you're welcome!

> this is my first PR of DefinitelyTyped, your reviews is so helped.

If you'd like to help with future reviews, add your name and GitHub alias below mine, like this:

```tsx
// Definitions by: Brendan Forster <https://github.com/shiftkey>
// Definitions by: Colin Luo <https://github.com/luozhihua>
```

I'm no longer reliant on `klawSync`, and it seems like you're more across the details than myself!

**Review by shiftkey — APPROVED** [review]

File: types/klaw-sync/index.d.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
 // Type definitions for klaw-sync 2.0
 // Project: https://github.com/manidlou/node-klaw-sync
 // Definitions by: Brendan Forster <https://github.com/shiftkey>
-// Definitions by: Colin Luo <https://github.com/luozhihua>
```
Comment (shiftkey) [inline types/klaw-sync/index.d.ts]: 😢

**Pull request merged**